Library/Formula/flex.rb

```diff
@@ -13,6 +13,8 @@ class Flex < Formula
   keg_only :provided_by_osx, "Some formulae require a newer version of flex."
 
   depends_on "gettext"
+  depends_on "bison" => :build if OS.linux?
+  depends_on "homebrew/dupes/m4" => :build if OS.linux?
```

Comment:
If `bison` depends on `m4` and `flex` depends on `bison`, then this dependency on `m4` is not needed.